### DISCUSSION TITLE
Grouping Access Modifiers

### DISCUSSION THREAD

**Discussion by osamakawish** [dotnet/csharplang#5162] [Language Ideas] [answered]
With the file-scoped namespace declarations coming up in C#10, I'd like it if access modifiers can be cleaned up as well. Though it's like C++'s notation, but instead of colons, you'd be using semi-colons.

Like:
```c#
public class A {
    private;
    int x;
    double dbl;
    string method(int yada) {yada.Tostring();}

    public;
    void DoStuff() {}
}
```

Criticisms welcome ofc.

**Comment by theunrepentantgeek** [comment]
That makes the visibility modifiers look like  discrete statements, unrelated to the code before or after.

More importantly though ...

One of the original precepts used in the design of C# was explicitness, especially where doing so would reduce the occurrence of bugs. This is why, for example, **virtual** is not the default (unlike Java).

Being explicit about visibility is similar - a small facet of design that reduces bug rates.

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/1021.
